src/planner.rs

```diff
@@ -18,12 +15,9 @@ impl RunArg {
     pub fn fetch<'a, U, F>(&'a self, f: F) -> U
         where F: FnOnce(&'a World) -> U
     {
-        let pulse = self.pulse.borrow_mut().take()
-                        .expect("fetch may only be called once.");
-        let u = f(&self.world);
-        pulse.pulse();
-        u
+        f(&self.world)
```

Comment:
You seem to be missing the point of the whole `fetch` phase here. If the code could just call `f`, we wouldn't need that phase in the first place! Here is why it's needed:
  - we don't want multiple systems to lock the storages they need to work on, since that is a recipe for deadlocks
  - thus, we separate the system execution into 2 phases: fetch and processing. The `fetch` phase execution is serialized
  - once a system finishes the `fetch` phase, it "detaches" from the planner and continues execution (the actual data processing) on its own, in parallel with all other things

I don't see how simply switching from threadpools to future cpu pools solves that.